src/js/testem-component.js

```diff
@@ -340,36 +333,19 @@ gpii.testem.generateUniqueDirName = function (basePath, prefix, suffix) {
  * @param that - The component itself.
  */
 gpii.testem.generateCoverageReportIfNeeded = function (that) {
-    var promise = fluid.promise();
-
     if (that.options.generateCoverageReport) {
-        promise.then(function () { fluid.log("Finished coverage report...");});
         try {
-            var commandSegments = [that.options.istanbulCmd, "report --root", fluid.module.resolvePath(that.options.coverageDir), "--dir", that.options.reportsDir, "text-summary html json-summary"];
-            var command = commandSegments.join(" ");
-
-            exec(command, { cwd: that.options.cwd }, function (error, stdout, stderr) {
-                fluid.log(stdout);
-                fluid.log(stderr);
-                if (error) {
-                    fluid.log("Error generating coverage report:", error);
-                    promise.reject(error);
-                }
-                else {
-                    fluid.log("Created coverage report in '", that.options.reportsDir, "'...");
-                    promise.resolve();
-                }
-            });
+            var commandSegments = ["report", "--root", fluid.module.resolvePath(that.options.coverageDir), "--dir", that.options.reportsDir, "text-summary",  "html", "json-summary"];
+            cli.runToCompletion(commandSegments);
+            fluid.log("Created coverage report in '", that.options.reportsDir, "'...");
         }
         catch (error) {
-            promise.reject(error);
+            fluid.fail(error);
```

Comment:
What was the reason for converting this reject into a fail?

Reply:
My previous approach called istanbul as an external command asynchronously, hence the promise.  The current approach loads istanbul's `cli` module and calls that synchronously.